javascript/components/FormInput.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+export default class ContactField extends Component {
+  render () {
+    let formElt;
+    const { inputType, inputValue, handleChange, fieldName, options } = this.props;
+    if (inputType === 'dropdown') {
+      formElt = (
+        <select name={this.props.fieldName.toLowerCase().split(' ')[0]} onChange={handleChange}>
+          {options.map((option, idx) => (
+            <option value={option.toLowerCase()} key={idx}>{option}</option>
+          ))}
+        </select>
+      );
+    } else if (inputType === 'textarea') {
+      formElt = (
+        <textarea value={inputValue} onChange={handleChange} name={this.props.fieldName.toLowerCase().split(' ')[0]}></textarea>
+      );
+    } else {
+      formElt = (
+        <input type={inputType} value={inputValue} onChange={handleChange} name={this.props.fieldName.toLowerCase().split(' ')[0]} />
+      );
+    }
+
+    return (
+      <div>
+        <label>
```

Comment:
As it stands, this <label> won't be associated with the input. In order for it to be associated, it needs to either be a parent of the element or the element needs to have an `id` prop and the `<label>` needs to have an `htmlFor={elementID}` prop.

Reply:
Fixed! Thanks.